testsuite/tests/lib-sys/rename.reference

```diff
@@ -2,3 +2,5 @@ Rename to nonexisting file: passed
 Rename to existing file: passed
 Renaming a nonexisting file: fails as expected
 Renaming to a nonexisting directory: fails as expected
+Rename directory to a nonexisting directory: passed
+Rename a nonexisting directory: fails as expected
```

Comment:
I think it would be worth testing renaming to an existing directory as well. There are some really weird corner cases (especially involving hardlinks) with filesystems, but just the simple case of ensuring that an existing target directory that contains a file with normal permissions is removed would be worthwhile.

Reply:
OK :+1:  5370e4a adds a negative test to rename to an existing, non-empty target directory.